backend/src/hmc-api/parser.ts

```diff
@@ -125,3 +125,125 @@ export function parse<Fields extends Record<string, string>>(
 
     return { ok: true, warnings, records };
 }
+
+export const enum BoomiWarningTag {
+    ExtraneousEntryValues = "ExtraneousEntryValues",
+    InsufficientEntryValues = "InsufficientEntryValues",
+    MissingSentinel = "MissingSentinel",
+}
+
+export type BoomiWarning =
+    | {
+          tag: BoomiWarningTag.ExtraneousEntryValues;
+          fields_desired: number;
+          fields_given: number;
+      }
+    | {
+          tag: BoomiWarningTag.InsufficientEntryValues;
+          fields_desired: number;
+          fields_given: number;
+      }
+    | {
+          tag: BoomiWarningTag.MissingSentinel;
+      };
+
+export const enum BoomiError {
+    MalformedHeader = "MalformedHeader",
+}
+
+export type BoomiResult<K extends string> =
+    | {
+          ok: true;
+          records: Record<K, string>[];
+          warnings: BoomiWarning[];
+      }
+    | {
+          ok: false;
+          error: BoomiError;
+      };
+
+/**
+ *
+ * @param fields A list containing the name of each field that the columns in the
+ *  database correspond to. A null value means that the column at that index should be skipped.
+ *  SAFETY: This list must contain at least one of every possible
+ *  non-null value of the type  Fields[number].
+ * @param data A string containing a Boomi database dump.
+ * @returns If parsing was successful, a result containing the list of all valid
+ *  database entries parsed into records according to the fields parameter,
+ *  and a list of warnings. If unsuccessful, returns the fatal error.
+ */
+export function parseBoomi<K extends string, Fields extends (K | null)[]>(
+    fields: Fields,
+    data: string,
+): BoomiResult<NonNullable<Fields[number]>> {
+    type EntryRecord = Record<NonNullable<Fields[number]>, string>;
+
+    const out_start = data.match(/OUT_START\|\d+\|\@\|/);
```

Comment:
I personally think it would be nice if this were slightly more strict-- say, include the `|@|` before the `OUT_START` in the regex pattern:

```typescript
data.match(/\|@\|OUT_START\|\d\|\@\|/);
```

(by the way, do we know what the meaning of the `\d` term in the db dump is?)